### PR TITLE
[WIP] extract client code from view

### PR DESCRIPTION
**Pull request by wacii**
Putting this up here more for discussion than anything. Is this the sort of code we want in the app?

The client is extracted into a class, which the view components can subscribe to for updates.

### CODE REVIEW

**Comment by knod** [comment]
Paraphrasing from Slack:
I like the idea of extracting into a class! It sounds like we'll save Redux for where we need it, partly to help show its clear use cases.

File: src/ClientModel.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { CLIENT_DEFAULTS } from './utils/CLIENT_DEFAULTS';
+
+export default class ClientModel {
+  constructor() {
+    this.listeners = [];
+    this.data = cloneDeep(CLIENT_DEFAULTS);
+    this.default = cloneDeep(CLIENT_DEFAULTS);
+    this.userChanged = {};
+    this.housing = this.data.current.housing;
+    this.isBlocking = false;
```
Comment (turnerhayes) [inline src/ClientModel.js:17]: What does `isBlocking` mean? I had thought it had something to do with blocking navigation from the visit page. Does it mean something in the context of the client?

Comment (knod) [reply src/ClientModel.js:17]: Not sure if it specifically needs to be in here. When the client has been changed, the navigation is blocked, that's all. We could possibly change this to `.isDirty` or something if we're going to put it in the client object.

We could possibly just check if `.userChanged` has any properties. I'm not sure which is worse, lol. Though I could see a future in which we check if the current values are different than the default values and set `.isDirty` based on that.

Comment (turnerhayes) [reply src/ClientModel.js:17]: I see. Yeah, I think the name `isBlocking` makes sense in VisitPage but not in the ClientModel. If it's in the ClientModel, it should probably be `.isDirty`. That could be a member property that's updated every time updateClientValue is called.

Comment (knod) [reply src/ClientModel.js:17]: That said, I personally don't count that as a blocking change here.

**Pull request closed without merging**